test/jdk/javax/swing/JFileChooser/6798062/bug6798062.java

```diff
@@ -21,73 +21,93 @@
-
-    private final JButton btnStart = new JButton("Start");
-
-    private final JButton btnStop = new JButton("Stop");
-
-    private final JButton btnGC = new JButton("Run System.gc()");
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JSlider;
+import javax.swing.JTextField;
+import javax.swing.SwingUtilities;
+
+public class bug6798062 {
+
+    private static final String INSTRUCTIONS = """
+            The test is suitable only for Windows.
+
+            1. Create a link
+            2. Copy path to the link into TextField
+            3. Run the Windows Task Manager. Select the Processes tab and find the java process
+            4. Press the Start button in the test window
+            5. Wait several minutes and observe in the Windows Task Manager
+               that Memory Usage of java process is not increasing
+            If memory usage is increasing, click Fail else click Pass .  """;
+
+    private static JSlider slider;
+    private static JTextField tfLink;
+    private static JButton btnStart;
+    private static JButton btnStop;
+    private static JButton btnGC;
 
     private ShellFolder folder;
-
     private Thread thread;
 
-    public static void main(String[] args) {
+    public static void main(String[] args) throws Exception {
+         PassFailJFrame.builder()
+                .title("JFileChooser Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows(10)
+                .columns(35)
+                .testUI(bug6798062::createUI)
+                .build()
+                .awaitAndCheck();
+    }
+
+    private static JFrame createUI() {
+        slider = new JSlider(0, 100);
+        tfLink = new JTextField();
+        btnStart = new JButton("Start");
+        btnStop = new JButton("Stop");
+        btnGC = new JButton("Run System.gc()");
         JFrame frame = new JFrame("bug6798062");
 
         frame.setSize(400, 300);
-        frame.setLocationRelativeTo(null);
         frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
         frame.add(new bug6798062().initialize());
 
-        frame.setVisible(true);
-    }
-
-    public void init() {
-        add(initialize());
+        return frame;
     }
 
     private JComponent initialize() {
         if (OSInfo.getOSType() != OSInfo.OSType.WINDOWS) {
             return new JLabel("The test is suitable only for Windows");
         }
 
-        String tempDir = System.getProperty("java.io.tmpdir");
-
-        if (tempDir.length() == 0) { // 'java.io.tmpdir' isn't guaranteed to be defined
-            tempDir = System.getProperty("user.home");
-        }
-
-        System.out.println("Temp directory: " + tempDir);
-
         try {
-            folder = ShellFolder.getShellFolder(new File(tempDir));
+            folder = ShellFolder.getShellFolder(new File("."));
```

Comment:
I wonder if it was expected that there are files in the directory. Like yes, so I could be wrong. I assumed files were created but no file is created, instead the files in this directory is listed. The expected result is that each file in the directory creates a new `ShellFolder` instance.

Reply:
One way to handle it is to create a number of files in the current directory, the `scratch` directory that jtreg provides and then rely on jtreg to remove it. Or to create a temp directory in the current directory and create files there.

We have no control over how many files are in the temporary directory, however, in most cases it's not empty.

Reply:
> I think we should revert back to "java.io.tmpdir" for this test sprint and figure out the fix for the tmpdir in JDK-8146446...Any objection?

Sounds reasonable.